Source/EasyNetQ/RabbitAdvancedBus.cs

```diff
@@ -88,15 +89,40 @@ AdvancedBusEventHandlers advancedBusEventHandlers
         produceDelegate = producePipelineBuilder.Use(_ => PublishInternalAsync).Build();
     }
 
+    /// <inheritdoc />
+    [Obsolete("IsConnected is deprecated because it is misleading. Please use GetConnectionStatus instead")]
+    public bool IsConnected =>
+        (from PersistentConnectionType type in Enum.GetValues(typeof(PersistentConnectionType)) select GetConnection(type))
+        .All(connection => connection.Status.State == PersistentConnectionState.Connected);
+
+    /// <inheritdoc />
+    [Obsolete("IsConnected is deprecated because it is misleading. Please use GetConnectionStatus instead")]
+    public async Task ConnectAsync(CancellationToken cancellationToken = default)
+    {
+        foreach (PersistentConnectionType type in Enum.GetValues(typeof(PersistentConnectionType)))
```

Comment:
why not tu use generic overload of `GetValues`? https://learn.microsoft.com/en-us/dotnet/api/system.enum.getvalues?view=net-7.0#system-enum-getvalues-1

Reply:
https://learn.microsoft.com/en-us/dotnet/api/system.enum.getvalues?view=netstandard-2.0#system-enum-getvalues-1 only non-generic version is available for netstandard-2.